src/gpuutils.py

```diff
@@ -0,0 +1,87 @@
+"""Utilities for measuring GPU performance."""
+
+from typing import Optional
+
+from logging import getLogger
+
+from pynvml import *
+import psutil
+
+import pandas as pd
+
+import time
+
+
+logger = getLogger(__file__)
+
+
+def gather_cpu_gpu_metrics(
+    interval: int | pd.Timedelta = 5,
+    iterations: int = 12,
+    *,
```

Comment:
I've never seen this before... is this like **kwargs?

Reply:
It says that arguments that follow it can only be passed by name. It's designed for functions that have a large number of args where it can be confusing what is what when you pass them all by position only.